src/evergreen/api.py

```diff
@@ -288,15 +288,19 @@ def __init__(
         """Create an Evergreen Api object."""
         super(_ProjectApi, self).__init__(api_server, auth, timeout)
 
-    def all_projects(self) -> List[Project]:
+    def all_projects(self, project_filter_fn: Optional[Callable] = None) -> List[Project]:
         """
         Get all projects in evergreen.
 
+        :param project_filter_fn: function to filter projects, should accept a project_id argument.
         :return: List of all projects in evergreen.
         """
         url = self._create_url("/projects")
         project_list = self._paginate(url)
-        return [Project(project, self) for project in project_list]  # type: ignore[arg-type]
+        projects = [Project(project, self) for project in project_list]  # type: ignore[arg-type]
```

Comment:
[nit] You could use a default filter that always returned `True` to clean this up a bit:
```
if project_filter_fn is None:
    project_filter_fn = lambda _: True
return [Project(project, self) for project in project_list if project_filter_fn(project)]
```

Reply:
I tried this, but flake8 doesn't like lambdas (`E731 do not assign a lambda expression, use a def`), and black doesn't like single-line method definitions, and pydocstyle requires a docstring (`Function is missing a type annotation`), so this ends up being less concise when I fix all those issues.